controllers/managedocs_controller.go

```diff
@@ -296,6 +315,11 @@ func (r *ManagedOCSReconciler) SetupWithManager(mgr ctrl.Manager) error {
 			enqueueManangedOCSRequest,
 			csvPredicates,
 		).
+		Watches(
+			&source.Kind{Type: &corev1.Service{}},
+			enqueueManangedOCSRequest,
+			servicePredicate,
+		).
```

Comment:
You don't need a manual watch, the `Owns` watch will cover this resource. I will suggest you should take a look at the implementation of `Owns` for better understanding

Reply:
Removed